src/main/java/org/ibp/api/java/impl/middleware/name/validator/GermplasmNameTypeValidator.java

```diff
@@ -0,0 +1,83 @@
+package org.ibp.api.java.impl.middleware.name.validator;
+
+import org.apache.commons.lang3.StringUtils;
+import org.generationcp.middleware.api.nametype.GermplasmNameTypeDTO;
+import org.generationcp.middleware.api.nametype.GermplasmNameTypeRequestDTO;
+import org.generationcp.middleware.api.nametype.GermplasmNameTypeService;
+import org.ibp.api.exception.ApiRequestValidationException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.MapBindingResult;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@Component
+public class GermplasmNameTypeValidator {
+
+	private static final Integer CODE_OR_NAME_MAX_LENGTH = 50;
+	private static final Integer DESCRIPTION_MAX_LENGTH = 255;
+	private BindingResult errors;
+
+	@Autowired
+	private GermplasmNameTypeService germplasmNameTypeService;
+
+	public void validate(final GermplasmNameTypeRequestDTO germplasmNameTypeRequestDTO) {
```

Comment:
There is no unit test for this function, I will include it as part of 18.4 @cuenyad